src/smbprotocol/connection.py

```diff
@@ -947,6 +947,7 @@ def disconnect(self, close=True, timeout=None):
 
         log.info("Disconnecting transport connection")
         self.transport.close()
+        self._t_worker.join(timeout=2)
```

Comment:
I did a short wait. mostly to help with testing.

For real world troubleshooting, where we should check and see if the thread is still running after the join and maybe log a warning ... or raise an exception.

For this PR, I wanted to keep the scope only to help with automated tests